pkg/auth/context.go

```diff
@@ -0,0 +1,106 @@
+// Package auth provides authentication and authorization utilities.
+package auth
+
+import (
+	"context"
+	"errors"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+// IdentityContextKey is the key used to store Identity in the request context.
+// This provides type-safe context storage and retrieval for authenticated identities.
+//
+// Using an empty struct as the key prevents collisions with other context keys,
+// as each empty struct type is distinct even if they have the same name in different packages.
+type IdentityContextKey struct{}
```

Comment:
I noticed that `pkg/vmcp/auth/context_test.go` (99 lines) was deleted. Were those tests migrated elsewhere?